File: result/repair/starchat/java2python/java_repair/PROGRAM_OCTAL_DECIMAL_CONVERSION.java
  int f_gold ( int n ) {
        int num = n, dec_value = 0, base = 1;
        while (temp){
            int last_digit = num % 10;
            num /= 10;
            dec_value += last_digit * base;
            base *= 8;
        }
        return dec_value;
    }

File: result/translate/starchat/java2python/python/PROGRAM_OCTAL_DECIMAL_CONVERSION.py
def f_gold(n):
        num = n
        dec_value = 0
        base = 1
        temp = num
        while temp > 0:
            last_digit = temp % 10
            temp = temp // 10
            dec_value += last_digit * base
            base = base * 8
        return dec_value